sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlockBlobClient.java

```diff
@@ -137,20 +137,9 @@ public BlockBlobItem upload(InputStream data, long length) throws IOException {
      */
     public Response<BlockBlobItem> uploadWithResponse(InputStream data, long length, BlobHTTPHeaders headers,
                                                       Metadata metadata, BlobAccessConditions accessConditions, Duration timeout, Context context) throws IOException {
-        Flux<ByteBuffer> fbb = Flux.range(0, (int) Math.ceil((double) length / (double) BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE))
-            .map(i -> i * BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE)
-            .concatMap(pos -> Mono.fromCallable(() -> {
-                long count = pos + BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE > length ? length - pos : BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE;
-                byte[] cache = new byte[(int) count];
-                int read = 0;
-                while (read < count) {
-                    read += data.read(cache, read, (int) count - read);
-                }
-                return ByteBuffer.wrap(cache);
-            }));
-
+        Flux<ByteBuffer> fbb = Utility.convertStreamToByteBuffer(data, length, BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE);
         Mono<Response<BlockBlobItem>> upload = blockBlobAsyncClient
-            .uploadWithResponse(fbb.subscribeOn(Schedulers.elastic()), length, headers, metadata, accessConditions, context);
+            .uploadWithResponse(fbb, length, headers, metadata, accessConditions, context);
```

Comment:
Did we mean to remove the `subscribeOn` here?

Reply:
It removed because previous implementation in azure core. 
Current way of taking request body does not affect by this. Will add it back.